From cb0ba7dffaa8dd4d3c0015c60620e13dc72fbfef Mon Sep 17 00:00:00 2001
From: zhoulei <zhoulei5@xiaomi.com>
Date: Wed, 3 Jan 2018 18:34:02 +0800
Subject: [PATCH 36/43] [AlarmManager]Use the correct func interface for send
 msg args

the ALMS used the obtainMessage(int what, Object obj) to
send REPORT_ALARMS_ACTIVE msg to mHandler,and used msg.arg1
in the handleMessage func, the func interface used incorrect

Change-Id: Idc40b422cf24030ac824154ed1682453dbbffa4c
Signed-off-by: zhoulei <zhoulei5@xiaomi.com>
---
 .../core/java/com/android/server/AlarmManagerService.java     | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/AlarmManagerService.java b/services/core/java/com/android/server/AlarmManagerService.java
index f79a51b13af..526b130af28 100644
--- a/services/core/java/com/android/server/AlarmManagerService.java
+++ b/services/core/java/com/android/server/AlarmManagerService.java
@@ -4073,7 +4073,7 @@ class AlarmManagerService extends SystemService {
                 Slog.d(TAG, "mBroadcastRefCount -> " + mBroadcastRefCount);
             }
             if (mBroadcastRefCount == 0) {
-                mHandler.obtainMessage(AlarmHandler.REPORT_ALARMS_ACTIVE, 0).sendToTarget();
+                mHandler.obtainMessage(AlarmHandler.REPORT_ALARMS_ACTIVE, null, 0).sendToTarget();
                 mWakeLock.release();
                 if (mInFlight.size() > 0) {
                     mLog.w("Finished all dispatches with " + mInFlight.size()
@@ -4237,7 +4237,7 @@ class AlarmManagerService extends SystemService {
                         alarm.type, alarm.statsTag, (alarm.operation == null) ? alarm.uid : -1,
                         true);
                 mWakeLock.acquire();
-                mHandler.obtainMessage(AlarmHandler.REPORT_ALARMS_ACTIVE, 1).sendToTarget();
+                mHandler.obtainMessage(AlarmHandler.REPORT_ALARMS_ACTIVE, null, 1).sendToTarget();
             }
             final InFlight inflight = new InFlight(AlarmManagerService.this,
                     alarm.operation, alarm.listener, alarm.workSource, alarm.uid,
-- 
2.17.1

